File: Proxy/Java/src/proxy/InternetProxy.java
package proxy;

import java.util.Arrays;
import java.util.List;

public class InternetProxy implements ISP {
    private List<String> blockSites = Arrays.asList("facebook","twitter");
    @Override
    public String serverSite(String url) {
        logSite(url);
        if(blockSites.contains(url)){
            return "\nthis site is blocked\n";
        }
        return  new Etisalat().serverSite(url);
    }
    public void logSite(String url){
        System.out.printf(" [%d] %s",System.currentTimeMillis(),url);
    }
}

File: Proxy/Python/InternetProxy.py
# -*- coding: utf-8 -*-
"""
Created on Mon May 30 20:26:51 2022

@author: Mahmoud Saeed
"""
from ISP import ISP
from Etisalat import Etisalat

class InternetProxy(ISP):
    blockList = ['facebook' , 'twitter']
    
    def serverSite(self , url):
        self.logSite(url)
        if url in self.blockList:
            return "this site is blocked"
        else:
            return Etisalat().serverSite(url)
    
    
    def logSite(self , url):
        print("log into {}".format(url))